### PR TITLE
Add use case labels and Update saving data use case

### LINKED ISSUE #112: Update saving data use case
AC:
1. Update statement 1 of saving data use case to be less implementation-focused. 
Statement 1: User enters a valid command that alters data (client add, client delete, client note add, etc).
2. Add numbering for use cases.

### PR DESCRIPTION
**Pull request by raysonkoh**
Fixes #112

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -329,26 +329,26 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-**Use case: Saving data**
+**UC3 - Saving data**
 
 **MSS**
 
-1.  User enters a valid command that alters data (`client add`, `client delete`, `client note add`, etc).
+1.  User enters a valid command that alters data (eg <ins>`adding a client (UC1)`</ins>).
```
Comment (tankangliang) [inline docs/DeveloperGuide.md]: Minor point, but can this be changed to "E.g." instead of "eg" with the period.

**Review by tankangliang — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (qwoprocks) [reply docs/DeveloperGuide.md]: Just to add on, I don't think the \<ins\> tag should be used here, as markup defines it as inserting something? e.g. [here](https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_del_css). \<u\> seems more appropriate if the intention is to underline the text.

Comment (raysonkoh) [reply docs/DeveloperGuide.md]: Okay I have made the changes, can you guys check again?

Comment (qwoprocks) [reply docs/DeveloperGuide.md]: LGTM

**Review by tankangliang — APPROVED** [review]
LGTM

**Pull request merged**